.ci/azure-pipelines/tutorials.yaml

```diff
@@ -11,6 +11,9 @@ jobs:
       CMAKE_CXX_FLAGS: '-Wall -Wextra -Wabi'
       EXCLUDE_TUTORIALS: 'davidsdk,ensenso_cameras,gpu'
     steps:
+      - checkout: self
+        # find the commit hash on a quick non-forced update too
```

Comment:
Could you please explain this a bit more? How does depths > 1 help during CI?

Reply:
If I push N commits one after another, then the commit hash given to the managing process might not be available in the depth, between the fetch and checkout

Reply:
Still don't get, but ok, does not hurt I suppose.

Reply:
1. Someone pushes to a branch
2. The CI gets the commit hash and starts searching for a free resource
3. Once it procures the compute resource, it starts downloading the docker image
4. Same branch gets updated
5. Then the CI starts cloning the repo with depth=1
6. CI fails to checkout the commit since it doesn't exist in the shallow history

This is the simplest race condition that can exist for a CI. Other scenarios involve force updates, merges, etc. CI doesn't stop the test on an update when it's configured to run per commit (eg: master) so this setting doesn't remove the issue completely

Reply:
Now this makes sense, thanks!

Reply:
It may become possible to shallow-clone a specific commit in future though: https://stackoverflow.com/a/43136160/1525865